local-cli/runAndroid/runAndroid.js

```diff
@@ -92,17 +92,20 @@ function buildAndRun(args, reject) {
   }
 
   try {
-    const packageName = fs.readFileSync(
-      'app/src/main/AndroidManifest.xml',
+    const manifest = fs.readFileSync(
+      'app/build/intermediates/manifests/full/debug/AndroidManifest.xml',
       'utf8'
-    ).match(/package="(.+?)"/)[1];
+    );
+
+    const applicationId = manifest.match(/package="(.+?)"/)[1];
+    const packageName = manifest.match(/android:name="(.+?\.MainActivity)"/)[1];
```

Comment:
yes this match could potentially fail, but that's also true for the other matches in the file. I'm not too much into Android and gradle stuff, but in contrast to the original version I'm referring to the AndroidManifest.xml generated by the build process, so the output should be more stable than the source AndroidManifest.